src/main/java/com/google/api/gax/grpc/ApiCallSettingsTyped.java

```diff
@@ -0,0 +1,92 @@
+package com.google.api.gax.grpc;
+
+import com.google.api.gax.core.RetrySettings;
+import com.google.common.collect.ImmutableSet;
+
+import io.grpc.ManagedChannel;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Package-private: Use other concrete settings classes instead of this class from outside of
+ * the package.
+ * A settings class with generic typing which can be used to configure an API method or create
+ * the method callable object. This class can be used as the base class that other concrete
+ * call settings classes inherit from.
+ */
+class ApiCallSettingsTyped<RequestT, ResponseT> extends ApiCallSettings {
+
+  private final MethodDescriptor<RequestT, ResponseT> methodDescriptor;
+
+  public MethodDescriptor<RequestT, ResponseT> getMethodDescriptor() {
+    return methodDescriptor;
+  }
+
+  public ApiCallSettingsTyped(Set<Status.Code> retryableCodes,
```

Comment:
Make protected. We don't want non-derived classes in this package creating instances.